### DISCUSSION TITLE
cosign private rekor sever

### DISCUSSION THREAD

**Discussion by nissessenap** [fluxcd/flux2#3309] [Proposals]
Today OCIRepository supports keyles verification of the public sigstore server using:

```.yaml
apiVersion: source.toolkit.fluxcd.io/v1beta2
kind: OCIRepository
metadata:
  name: cert-manager
spec:
  verify:
    provider: cosign
``` 

This is great, but I can see that people don't want to use the public sever and instead want to use a locally hosted one.

It would be a nice addition to add support to customize the rekor endpoint being used.

**Comment by KarstenSiemer** [comment]
I agree. Due to airgapped clusters, the public server is unreachable

**Comment by nissessenap** [comment]
I haven't looked at flux features in a long long time sadly. But it would also be resonable to have this provider as oci and pull the attesetation from your container registry. Probably the solution most people go with instead of hosting private rekors (it is a pain).
